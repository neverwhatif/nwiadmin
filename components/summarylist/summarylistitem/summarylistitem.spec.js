import React from 'react';
import { shallow } from 'enzyme';

import SummaryListItem from './index';

describe('SummaryListItem', () => {
    const component = shallow(<SummaryListItem term="Lorem" def="Ipsum" />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render a term', () => {
        expect(component.find('dt').text()).toBe('Lorem');
    });
    it('should render a def', () => {
        expect(component.find('dd').text()).toBe('Ipsum');
    });
});
