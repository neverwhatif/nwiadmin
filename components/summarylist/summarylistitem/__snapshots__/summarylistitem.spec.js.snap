// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryListItem should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SummaryListItem
    def="Ipsum"
    term="Lorem"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner"
      >
        <dt
          className="term"
        >
          Lorem
        </dt>
        <dd
          className="def"
        >
          Ipsum
        </dd>
      </div>,
      "className": "root",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <dt
            className="term"
          >
            Lorem
          </dt>,
          <dd
            className="def"
          >
            Ipsum
          </dd>,
        ],
        "className": "inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "term",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "dt",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ipsum",
            "className": "def",
          },
          "ref": null,
          "rendered": "Ipsum",
          "type": "dd",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner"
        >
          <dt
            className="term"
          >
            Lorem
          </dt>
          <dd
            className="def"
          >
            Ipsum
          </dd>
        </div>,
        "className": "root",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <dt
              className="term"
            >
              Lorem
            </dt>,
            <dd
              className="def"
            >
              Ipsum
            </dd>,
          ],
          "className": "inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lorem",
              "className": "term",
            },
            "ref": null,
            "rendered": "Lorem",
            "type": "dt",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ipsum",
              "className": "def",
            },
            "ref": null,
            "rendered": "Ipsum",
            "type": "dd",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
